raiden/transfer/node.py

```diff
@@ -595,6 +613,21 @@ def handle_receive_secret_reveal(node_state, state_change):
     return subdispatch_to_paymenttask(node_state, state_change, secrethash)
 
 
+def handle_processed(node_state, state_change):
+    # TODO: imporve the complexity of this algorithm
```

Comment:
same here: Also I guess you should abstract the outer loop it into another function since this is doing the same as in `handle_delivered`